lib/components/narrative/save-trip-button.js

```diff
@@ -16,31 +18,32 @@ const SaveTripButton = ({
   loggedInUser,
   persistence
 }) => {
+  const intl = useIntl()
   // We are dealing with the following states:
   // 1. Persistence disabled => just return null
   // 2. User is not logged in => render something like: "Please sign in to save trip".
   // 3. itin cannot be monitored => disable the button with prompt and tooltip.
 
   let buttonDisabled
-  let buttonText
-  let tooltipText
-  let icon
+  let buttonTextId
+  let tooltipTextId
+  let iconType
 
   if (!persistence || !persistence.enabled) {
     return null
   } else if (!loggedInUser) {
     buttonDisabled = true
-    buttonText = 'Sign in to save trip'
-    icon = 'fa fa-lock'
-    tooltipText = 'Please sign in to save trip.'
+    buttonTextId = 'components.SaveTripButton.signInText'
+    iconType = 'lock'
+    tooltipTextId = 'components.SaveTripButton.signInTooltip'
```

Comment:
The [`@formatjs/cli`](https://formatjs.io/docs/tooling/cli) tool (which we may want to use in the future) isn't able to parse dynamically generated message IDs. So I recommend changing this as follows:

```suggestion
    buttonText = intl.formatMessage({ id: 'components.SaveTripButton.signInText' })
    iconType = 'lock'
    tooltipText = <FormattedMessage id='components.SaveTripButton.signInTooltip' />
```

Reply:
Updated in https://github.com/opentripplanner/otp-react-redux/pull/430/commits/b28ccbdecf85d6e381b8a03c27b0e2910bda849f.